server/src/bot.js

```diff
@@ -70,6 +71,108 @@ const askLocation = (response, convo) => {
   convo.next();
 }
 
+connection.hears(["tag", "filter"], ['direct_message'], (bot, message) => {
+  respondWithTags(bot, message);
+});
+
+const respondWithTags = (bot, message) => {
+  //find all tags
+  helper.listAllTags()
+  .then(allTags => {
+    //find all user tags
+    helper.listUserTags(message)
+    .then(res => {
+      //all user tags
+      let userTag = _.map(res, item => {
+        return item.tagId;
+      });
+      let attachments = [];
+
+      //how can i loop through the user tags
+      //if the user has the tag, have a delete button
+      //otherwise, have a button to add
+
+      allTags.forEach(({id, name}) => {
+        let addButton =  {
+          name: `addTag`,
+          text: `Add Tag`,
+          value: id,
+          type: `button`,
+          style: `primary`
+        }; 
+        let deleteButton = {
+          name: `deleteTag`, 
+          text: `Delete Tag`, 
+          value: id, 
+          type: `button`, 
+          style: `danger`,
+          confirm: {
+            title: `Are you sure?`,
+            text: `Confirmation to delete tag?`,
+            ok_text: `Yes, delete it!`,
+            dismiss_text: `No, don't delete!`
+          } 
+        };
+        //does tag(user tag) exist in tags(tag table)
+        let button = (userTag.indexOf(id) !== -1) ? deleteButton : addButton;
+            
+        let attachment = {
+          text: `${name}`,
+          callback_id: `userTag`,
+          attachment_type: `default`,
+          color: `#3AA3E3`,
+          actions: [button]
+        };
+        attachments.push(attachment);
+      });
+
+      let response = {
+        text: `Here are a list of your tags: `,
+        fallback: `Unable to show tags`,
+        color: `#3AA3E3`,
+        attachments: attachments
+      };
+
+      bot.reply(message, response);
+    });
+  });
+};
+
+connection.hears("location", ['direct_message'], (bot, message) => {
+  bot.reply(message, { 
+    text: `Would you like to *view* or *update* your current job search location?`,
+    attachments: [
+      {
+        text: `Choose to view or update`,
+        fallback: `You are unable to choose`,
```

Comment:
Would reword this to something else -- "This option is disabled" or something softer